server/spawn-vehicle.lua

```diff
@@ -13,46 +13,67 @@ end
 ---@param garageName string
 ---@param accessPointIndex integer
 ---@return number? netId
-lib.callback.register('qbx_garages:server:spawnVehicle', function (source, vehicleId, garageName, accessPointIndex)
+lib.callback.register('qbx_garages:server:spawnVehicle', function(source, vehicleId, garageName, accessPointIndex)
     local garage = Garages[garageName]
     local accessPoint = garage.accessPoints[accessPointIndex]
-    if #(GetEntityCoords(GetPlayerPed(source)) - accessPoint.coords.xyz) > 3 then
-        lib.print.error(string.format("player %s attempted to spawn a vehicle but was too far from the access point", source))
-        return
-    end
     local garageType = GetGarageType(garageName)
-
     local spawnCoords = accessPoint.spawn or accessPoint.coords
-    if Config.distanceCheck then
-        local vec3Coords = vec3(spawnCoords.x, spawnCoords.y, spawnCoords.z)
-        local nearbyVehicle = lib.getClosestVehicle(vec3Coords, Config.distanceCheck, false)
-        if nearbyVehicle then
-            exports.qbx_core:Notify(source, locale('error.no_space'), 'error')
-            return
-        end
-    end
+
+    local hookPayload = {
+        source = source,
+        vehicleId = vehicleId,
+        garage = garage,
+        garageName = garageName,
+        accessPointIndex = accessPointIndex,
```

Comment:
This seems like an internal field to garages. Do not expose it to hooks payload.